clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -30,7 +30,40 @@
  */
 public interface Time {
 
-    Time SYSTEM = new SystemTime();
+    //  A time implementation that uses the system clock and sleep call.
+    //  Use inline implementation to ensure that only one Time#SYSTEM exists in a program
+    Time SYSTEM = new Time() {
```

Comment:
I think rather than being an anonymous Time$1, it should still be a named SystemTime.
You could leave the existing class in-place and make it package-local, or make this an inner class.

Reply:
> You could leave the existing class in-place and make it package-local, or make this an inner class.

As `Timer` is a interface, inner class has public modifier by default. Hence +1 to make it package-private - `public class SystemTime` -> `class SystemTime`

Reply:
Thank you
I will modify `SystemTime` to be package private and provide singleton access